homeassistant/components/garmin_connect/sensor.py

```diff
@@ -109,7 +104,22 @@ def icon(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._state
+        if self.coordinator.data:
+            if self.coordinator.data[self._type]:
```

Comment:
Combine these two checks with `and`. Or even better invert the checks and combine them with `or` and then return None if true. Then we can outdent below.